src/connector/src/sink/postgres.rs

```diff
@@ -0,0 +1,626 @@
+                .iter()
+                .map(|i| schema.fields()[*i].data_type().to_pg_type())
+                .collect_vec();
+            let delete_sql = create_delete_sql(&schema, &config.table, &pk_indices);
+            Some(
+                client
+                    .prepare_typed(&delete_sql, &delete_types)
+                    .await
+                    .context("Failed to prepare delete statement")?,
+            )
+        };
+
+        let merge_statement = if is_append_only {
+            None
+        } else {
+            let merge_types = schema
+                .fields
+                .iter()
+                .map(|field| field.data_type().to_pg_type())
+                .collect_vec();
+            let merge_sql = create_merge_sql(&schema, &config.table, &pk_indices);
+            Some(
+                client
+                    .prepare_typed(&merge_sql, &merge_types)
+                    .await
+                    .context("Failed to prepare merge statement")?,
+            )
+        };
+
+        let writer = Self {
+            config,
+            pk_indices,
+            is_append_only,
+            client,
+            buffer: Buffer::new(),
+            insert_statement,
+            delete_statement,
+            merge_statement,
+        };
+        Ok(writer)
+    }
+
+    async fn flush(&mut self) -> Result<()> {
+        if self.is_append_only {
+            for chunk in self.buffer.drain() {
+                for (op, row) in chunk.rows() {
+                    match op {
+                        Op::Insert => {
+                            self.client
+                                .execute_raw(&self.insert_statement, row.iter())
+                                .await?;
+                        }
+                        Op::UpdateInsert | Op::Delete | Op::UpdateDelete => {
+                            debug_assert!(!self.is_append_only);
+                        }
+                    }
+                }
+            }
+        } else {
+            for chunk in self.buffer.drain() {
+                for (op, row) in chunk.rows() {
+                    match op {
+                        Op::Insert => {
+                            self.client
+                                .execute_raw(&self.insert_statement, row.iter())
+                                .await?;
+                        }
+                        Op::UpdateInsert => {
+                            // NOTE(kwannoel): Here we use `MERGE` rather than `UPDATE/INSERT` directly.
+                            // This is because the downstream db could have cleaned the old record,
+                            // in that case it needs to be `INSERTED` rather than UPDATED.
+                            // On the other hand, if the record is there, it should be `UPDATED`.
+                            self.client
+                                .execute_raw(self.merge_statement.as_ref().unwrap(), row.iter())
```

Comment:
Notice that we can directly iter on `row` here. That's because we implemented `ToSql` for `ScalarRefImpl` above.